prime-angular-frontend/src/app/modules/site-registration/shared/services/site-form-state.service.ts

```diff
@@ -161,9 +163,16 @@ export class SiteFormStateService extends AbstractFormState<Site> {
     }
 
     if (site.businessHours?.length) {
-      const array = this.hoursOperationForm.get('businessDays') as FormArray;
-      array.clear(); // Clear out existing indices
-      this.formUtilsService.formArrayPush(array, site.businessHours);
+      const array = [];
+      for (let index = 0; index < 7; index++) {
```

Comment:
Feels like we can drop this loop and use method chaining.  Something like `const businessDays = [...new Array(7)].reduce((days, dayOfWeek) => ..., []);`